File: 0485-max-consecutive-ones/0485-max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int temp = 0;
        int result = 0;
        for (int num : nums) {
            if(num == 1) temp++;
            else if(num == 0) {
                if (temp > result)
                    result = temp;
                temp = 0;
            }
        }
        if(temp > result) result = temp;
        return result;
    }
}

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.kt
class Solution {
fun findMaxConsecutiveOnes(nums: IntArray): Int {
    var temp =0;
    var max = 0;

    nums.forEach {
        if(it == 1){
            temp++
        } else{
            if(temp > max){
                max = temp
            }
            temp = 0
        }
    }
    if(temp > max){
        max = temp
    }
    return max
}
}